src/main/java/interaction/rule/NoInteractionRule.java

```diff
@@ -11,7 +12,17 @@
     public static final NoInteractionRule instance = new NoInteractionRule();
```

Comment:
what is this rule for? Is this normal or dev mistake case? If its developer error then you should throw unchecked exception instead of returning NoInteraction rule.

Reply:
it's normal, when `Interaction` isn't given `InteractionRule`, it won't produce any `Action`, `NoInteractionRule` return that kind of rule that return empty `Action` collection